Apache-Arrow-Flight-Manager-Tester/main.py

```diff
@@ -0,0 +1,116 @@
+import pprint
+from typing import Literal
+
+from pyarrow import flight, Schema
+from pyarrow._flight import FlightInfo, FlightClient, ActionType, FlightEndpoint
+
+
+# PyArrow Functions.
+def list_flights(flight_client: FlightClient) -> list[FlightInfo]:
+    response = flight_client.list_flights()
+
+    return list(response)
+
+
+def get_schema(flight_client: FlightClient, table_name: str) -> Schema:
+    upload_descriptor = flight.FlightDescriptor.for_path(table_name)
+    response = flight_client.get_schema(upload_descriptor)
+
+    return response.schema
+
+
+def do_action(flight_client: FlightClient, action_type: str, action_body: str) -> list[any]:
+    action_body_bytes = str.encode(action_body)
+    action = flight.Action(action_type, action_body_bytes)
+    response = flight_client.do_action(action)
+
+    return list(response)
+
+
+def list_actions(flight_client: FlightClient) -> list[ActionType]:
+    response = flight_client.list_actions()
+
+    return list(response)
+
+
+# Helper functions.
+def list_table_names(flight_client: FlightClient) -> list[str]:
+    flights = list_flights(flight_client)
+
+    return [table_name.decode("utf-8") for table_name in flights[0].descriptor.path]
+
+
+def initialize_database(flight_client: FlightClient, tables: list[str]) -> list[str]:
+    result = do_action(flight_client, "InitializeDatabase", ",".join(tables))[0]
+    decoded_result = result.body.to_pybytes().decode("utf-8")
+
+    return decoded_result.split(";")
+
+
+def execute_query(flight_client: FlightClient, query: str) -> None:
+    # Retrieve the flight info that describes how to execute the query.
+    query_descriptor = flight.FlightDescriptor.for_command(query)
+    flight_info: FlightInfo = flight_client.get_flight_info(query_descriptor)
+
+    # Use the flight endpoint in the returned flight info to execute the query.
+    endpoint: FlightEndpoint = flight_info.endpoints[0]
+    cloud_node_url = endpoint.locations[0]
+
+    cloud_client = flight.FlightClient(cloud_node_url)
+    response = cloud_client.do_get(endpoint.ticket)
+
+    for batch in response:
+        pprint.pprint(batch.data.to_pydict())
+
+
+def update_object_store(flight_client: flight.FlightClient, object_store_type: Literal["s3", "azureblobstorage"],
+                        arguments: list[str]) -> list[any]:
```

Comment:
Can the return type be made more specific? Looks like [`do_action()`](https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightClient.html#pyarrow.flight.FlightClient.do_action) always return the same [`Result`](https://arrow.apache.org/docs/python/generated/pyarrow.flight.Result.html) type.

Reply:
fIXED.